Fix no longer existing download url in graphviz

nazavode opened the pull request:
It looks like the Graphviz guys migrated their repo to a brand new platform that broke all the previously working download URLs. They are now providing source packages for version `2.40.1` only. After lurking around the web, I've found the missing package (`graphviz-2.38.0.tar.gz`) on the Fedora package repo. The file hash is the same.

nazavode commented:
Got it, yours looks like the optimal solution. I'm going to try adding the latest version right now to see what happens.

nazavode commented:
Update:
1. Their (only) provided source package is an unversioned tarball (https://graphviz.gitlab.io/pub/graphviz/stable/SOURCES/graphviz.tar.gz). Without any mean to tell which version contains, using it looks a bit scary.
2. Tried to pull the stable `2.40.1` version directly from their repo (tag `stable_release_2.40.1`) but it is broken (`autoreconf` fails due to missing `m4` input files).

adamjstewart commented:
Would you mind opening an issue with them and complaining about the lack of a stable download source?

There's nothing wrong with pulling from an unversioned tarball since we always confirm the checksum before installing. The only problem is that it will change in the future unless they provide us some stable download source.

nazavode commented:
> Would you mind opening an issue with them and complaining about the lack of a stable download source?

Sure, I'm going to investigate about it.

> There's nothing wrong with pulling from an unversioned tarball since we always confirm the checksum before installing. The only problem is that it will change in the future unless they provide us some stable download source.

It looks scary just because it is a bit suspicious that an ubiquitous package like that offers no proper way to retrieve stable releases. That tarball is the only one available, and it is more than a year old. Looking into it!

nazavode commented:
@adamjstewart @alalazo update:

- Updated `graphviz` to latest version via proper `git` tag.
- Cleaned up dependencies according to [official build instructions](https://graphviz.gitlab.io/_pages/Download/Download_source.html).
- Added `+expat` as variant (disabled by default, just like all remaining optional dependencies).
- Tested all variant combinations on `%gcc@5.4.0 arch=linux-ubuntu16.04-x86_64`.

alalazo commented on var/spack/repos/builtin/packages/graphviz/package.py:
```diff
@@ -23,16 +23,16 @@
 
-    version('2.38.0', '5b6a829b2ac94efcd5fa3c223ed6d3ae')
+    version('2.40.1', git='https://gitlab.com/graphviz/graphviz.git',
+            tag='stable_release_2.40.1')
```
Usually we try to have release versions that are tied to an hash, for security reasons. Does it make sense in this case to use the commit hash instead?

nazavode replied:
Yup!

alalazo commented on var/spack/repos/builtin/packages/graphviz/package.py:
```diff
@@ -96,28 +97,46 @@ class Graphviz(AutotoolsPackage):
+    depends_on('freetype', when='+pangocairo')
+    depends_on('glib', when='+pangocairo')
+    depends_on('fontconfig', when='+pangocairo')
+    depends_on('freetype', when='+pangocairo')
```
I think this line is repeated two times

nazavode commented:
@adamjstewart @alalazo added (and tested) some low hanging fruits as variants. Pls let me know if you prefer this to be in sync with HEAD before review.

alalazo commented:
@nazavode Just waiting for Travis to complete, then I'll merge this. MacOS builds have been a pita lately.

alalazo commented:
Thanks!
